### ISSUE TITLE
[Critical] : OTP Bypass leads to ACCOUNT TAKEOVER

### ISSUE THREAD

**Issue by CYBERDEVILZ** [P0intMaN/zbunker-website#119]
# Critical Security Issue
### Steps to reproduce
In the following video, I have shown how an attacker can abuse the OTP Authentication System and use it in his favor to gain access to any account using a web proxy like **BurpSuite** or **OWASP Zap**   

https://user-images.githubusercontent.com/55954313/137865614-e368b9ef-56f5-491b-a4e6-20f9e8bcc4dc.mp4

Pinging @ashutoshkrris  @P0intMaN

**Comment by CYBERDEVILZ** [comment]
@ashutoshkrris 
Instead of sending the response whether the OTP is correct or not, what you can do is, if the OTP is correct then redirect the user to a password-reset page, where the user will reset his password. But make sure to authenticate whether it is the correct user who has been redirected. One way to do is by using a CSRF token. And if the OTP is wrong, then you can give the user an alert saying the OTP is wrong.

**Comment by P0intMaN** [comment]
Interesting stuff @CYBERDEVILZ . So, you did an account takeover. Upon inspecting the code, I found that the OTP system is based on AJAX and essentially, the OTP is being validated by a response being sent by the server. Here, an attacker can spoof the validation by modifying the request.

So, If I had followed along correctly, what @CYBERDEVILZ did was you modified the OTP_mismatch to bypass this. 

One solution to this is to let Backend decide what happens when the OTP is validated and in this case, I agree with you to redirect the user to a separate password-reset page.
